File: bindings/python/tests/test_ema.py
from time import sleep

from EMA import (
    EMA_finalize,
    EMA_init,
    EMA_region_begin,
    EMA_region_define,
    EMA_region_end,
)


EMA_init()
region = EMA_region_define('region0')
EMA_region_begin(region)

print('sleep begin')
sleep(3)
print('sleep end')

EMA_region_end(region)

EMA_finalize()

File: bindings/rust/EMA/src/bin/test_ema.rs
fn main() {
    println!("Initializing EMA...");
    EMA::init().expect("Failed to init EMA!");

    let plugins = EMA::PluginList::new();
    println!("Found {} plugins:", plugins.size());

    for plugin in plugins.iter() {
        println!("\t{}", plugin.name());
    }

    let devices = EMA::DeviceList::new();
    println!("Found {} devices:", devices.size());

    for device in devices.iter() {
        println!("\t{}", device.name());
    }

    let filter = EMA::Filter::new("NVML");

    let region = EMA::Region::new("region", &filter);
    region.begin();
    std::thread::sleep(std::time::Duration::from_secs(5));
    region.end();

    filter.finalize().expect("Failed to finalize filter!");

    println!("Finalizing EMA...");
    EMA::finalize().expect("Failed to finalize EMA!");
}
